pycalphad/core/phase_rec.pyx

```diff
@@ -47,56 +47,11 @@ cdef public class PhaseRecord(object)[type PhaseRecordType, object PhaseRecordOb
     @cython.boundscheck(False)
```

Comment:
I added a short explanation to the docstrings of `CompositionSet` and `PhaseRecord`, essentially just copying from the text explaining this PR.